### PR TITLE
JS Build Decoupling from Django stack

### PR DESCRIPTION
**Pull request by rtibbles**
### Summary
Supersedes #3379 

from that PR:

  I replaced the webpack_json management command with a command line tool, build_tools/webpack_json.py, that allows you to initialize and build specified kolibri plugins, rather than building every single plugin (as the management command did).

image

…

### Reviewer guidance

 * A reviewer can attempt to run the command " python webpack_json.py" with either of the following options:
1. -plugin_file option = enter an input file with a list of kolibri plugins you'd like to build.
2. -plugins option = enter a python list of plugins, or a space-separated list of plugins to build.

### References
First step to addressing #950

----

### Contributor Checklist

- [ ] Contributor has fully tested the PR manually
- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label

### Reviewer Checklist

- [ ] Automated test coverage is satisfactory
- [ ] Reviewer has fully tested the PR manually
- [ ] PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- [ ] External dependencies files were updated (`yarn` and `pip`)
- [ ] Documentation is updated
- [ ] Link to diff of internal dependency change is included
- [ ] CHANGELOG.rst is updated for high-level changes
- [ ] Contributor is in AUTHORS.rst

### CODE REVIEW

File: build_tools/webpack_json.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import argparse
+import json
+import logging
+import os
+import sys
+
+from customize_build import load_plugins_from_file
+
+from kolibri.core.webpack.hooks import WebpackBundleHook
```
Comment (lyw07) [inline build_tools/webpack_json.py:9]: According to the buildkite error, I think we might need to call `set_env()` before doing this import

Comment (rtibbles) [reply build_tools/webpack_json.py:9]: Right, will check if there is a way to make this less of an issue.

Comment (rtibbles) [reply build_tools/webpack_json.py:9]: Have followed the suggestion from `cli.py` to move set_env behaviour into `kolibri.__init__`, so this is now working, and we should always be able to use kolibri as an imported library without special setup!

File: build_tools/webpack_json.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        else:
+                logger.info('Successfully initialized {}'.format(module))
+
+def main():
+    logger = logging.getLogger(__name__)
+    logging.StreamHandler(sys.stdout)
+    logger.setLevel(logging.INFO)
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument('--plugin_file', help='the filepath to which you\'d like to run plugins from', type=str, default=None)
```
Comment (lyw07) [inline build_tools/webpack_json.py:34]: Sorry I'm not sure if I understand this correctly, and I don't really know about webpack. So my understanding is that we can now enable any plugins we want using `python webpack_json.py --plugin_file/--plugins`. Is this usage only for dev mode? Thank you!

Comment (rtibbles) [reply build_tools/webpack_json.py:34]: Yes, although this command is currently invoked from inside the webpack build process. For now, this still uses the environment variable as the primary means for configuration of the list of build plugins, but these command arguments are here to allow more flexibility in the future.

If you want to test this out you can run `BUILD_TIME_PLUGINS=test.txt yarn run devserver ` where `test.txt` is the path to a file with a list of plugin module paths you want to build.

Comment (lyw07) [reply build_tools/webpack_json.py:34]: thanks for the info!

**Pull request merged**